plugins/serializers/template/README.md

```diff
@@ -0,0 +1,33 @@
+# Template Serializer
+
+The `template` output data format outputs metrics using an user defined go template.
+
+## Configuration
+
+```toml
+[[outputs.file]]
+  ## Files to write to, "stdout" is a specially handled file.
+  files = ["stdout", "/tmp/metrics.out"]
+
+  ## Data format to output.
+  ## Each data format has its own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  data_format = "template"
+
+  ## Go template which defines output format
+  template = '{{ .Tag "host" }} {{ .Field "available" }}'
+```
+
+### Batch mode
+
+When an output plugin emits multiple metrics in a batch fashion the template receives the
+array of metrics as the dot.
+
+```toml
+template = '''My metric names: 
+{{- range $index, $metric := . -}}
+{{if $index}}, {{ end }}{{ $metric.Name }}
+{{- end }}
+'''
+```
```

Comment:
Currently, the user cannot know if the serialization is done per-metric or as a batch as it depends on the output plugin. I would advocate for either a flag which says `batch_template` or something or two different templates. You then need to wrap/unwrap the metric(s) in the corresponding serialization method to fit the template IMO...

What do you think?

Reply:
Yeah, I agree that's confusing. Especially since the output plugins don't all clearly define what mode they use. Some of them let the user choose already though, and I'd like to add that user choice to exec/execd plugins too (#13654). I'm not sure what the best solution is. The flag way would be a bit weird to have to define 2 flags when the output plugin itself has one to switch to batch mode, and one for the serializer. Perhaps having a separate batch_template field in the config could work though. Let me think about it and try some stuff out while I fix the other notes you made.

Reply:
Okay, I split out a separate `batch_template`, and made fallbacks in both directions in case the 'wrong' template type was defined.